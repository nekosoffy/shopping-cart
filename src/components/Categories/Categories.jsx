import styles from './Categories.module.css';

export default function Categories() {
  const categories = ['DRESSES', 'TOPS', 'PLUS + CURVE', 'MEN']

  return (
    <section className={styles.categories}>
      <h2>SHOP CATEGORIES</h2>
      <div className={styles.background}></div>
      <nav>
        <ul className={styles["categories-wrapper"]}>
          {categories.map((item, i) => {
            return (
            <li key={i} className={styles[`img-container${i}`]}>
              <a href="">{item}</a>
            </li>
            )
          })}
        </ul>
      </nav>
    </section>
  );
}
