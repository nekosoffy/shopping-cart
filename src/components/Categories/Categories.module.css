.categories {
  display: grid;
  justify-items: center;
  gap: 30px;
  position: relative;
  margin-bottom: 30px;
  margin-top: 50px;

  /* Font-sizes */
  --heading: 3.2rem;
  --categories: 2.3rem;
}

.categories-wrapper {
  display: flex;
  gap: 50px;
}

.categories > h2 {
  font-family: 'Aileron', sans-serif;
  color: var(--smooth-dark-font);
  font-size: var(--heading);
}

.img-container {
  composes: img-container from "../Carousel/Carousel.module.css";
  height: 500px;
  width: 350px;
  margin-bottom: 30px;
  border-radius: 5px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'Lato', sans-serif;
  color: var(--light-font);
  font-size: var(--categories);
  pointer-events: none;
  transition: filter 0.5s ease;
}

.img-container:hover {
  filter: brightness(75%);
}

.img-container0 {
  composes: img-container;
  background: center / cover no-repeat url(../../assets/dresses.jpg);
}

.img-container1 {
  composes: img-container;
  background: center / cover no-repeat url(../../assets/tops.jpg);
}

.img-container2 {
  composes: img-container;
  background: center / cover no-repeat url(../../assets/curve.jpg);
}

.img-container3 {
  composes: img-container;
  background: center / cover no-repeat url(../../assets/men.jpg);
}

.img-container > a {
  padding: 20px;
  background-color: rgba(0, 0, 0, 0.734);
  text-align: center;
  width: 80%;
  pointer-events: auto;
  border-radius: 2px;
}

.background {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 60%;
  bottom: 0;
  background-color: rgb(255, 218, 224);
}