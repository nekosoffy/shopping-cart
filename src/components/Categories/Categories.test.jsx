import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Categories from './Categories';

describe('Categories component', () => {
  it('Matches the snapshot', () => {
    const { asFragment } = render(<Categories />);
    expect(asFragment()).toMatchSnapshot();
  });
});
