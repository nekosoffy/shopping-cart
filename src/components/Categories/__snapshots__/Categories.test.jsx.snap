// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Categories component > Matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="_categories_0f9d92"
  >
    <h2>
      SHOP CATEGORIES
    </h2>
    <div
      class="_background_0f9d92"
    />
    <nav>
      <ul
        class="_categories-wrapper_0f9d92"
      >
        <li
          class="_img-container0_0f9d92"
        >
          <a
            href=""
          >
            DRESSES
          </a>
        </li>
        <li
          class="_img-container1_0f9d92"
        >
          <a
            href=""
          >
            TOPS
          </a>
        </li>
        <li
          class="_img-container2_0f9d92"
        >
          <a
            href=""
          >
            PLUS + CURVE
          </a>
        </li>
        <li
          class="_img-container3_0f9d92"
        >
          <a
            href=""
          >
            MEN
          </a>
        </li>
      </ul>
    </nav>
  </section>
</DocumentFragment>
`;
